### ISSUE TITLE
Support for `mac1.metal`/`mac2.metal` instances

### ISSUE THREAD

**Issue by tecracer-theinen** [test-kitchen/kitchen-ec2#591]
## :frowning_person: Problem Statement

I want to test some Mac-specific implementations with AWS.

## :grey_question: Possible Solution

Add the necessary logic to select mac-AMIs and wire it to the corresponding types. 

As `mac`-instances need to be on a [dedicated host and are billed 24 hours minimum](https://aws.amazon.com/ec2/dedicated-hosts/pricing/), this needs some dedicated host logic (see my issue #590)

## :arrow_heading_up: Describe alternatives you've considered

No alternatives currently, as AMI selection and other logic is missing.

## :heavy_plus_sign: Additional context

I already have an almost-working implementation for this. Will do some cleanups and PR after #590

**Comment by mfortin** [comment]
What's the status ?
Host tenancy is one of the possible configuration, however there are alternatives, like using using a placement host_group_arn and license_configuration, can this also be supported ?

**Comment by hiSandog** [comment]
The scope would be easier to review if the first version defined a small acceptance contract for `mac1.metal`. I would call out the default behavior, the migration/backwards-compatibility story, and one or two cases that are intentionally out of scope so the implementation does not grow beyond the original request.

**Comment by tas50** [comment]
This shipped some time ago — closing as implemented.

**AMI selection** landed in #594 (`Add support for mac instances`), which added `lib/kitchen/driver/aws/standard_platform/macos.rb`. A platform named `macos` (optionally `macos-<version>`) resolves Amazon's macOS AMIs under owner `100343932686`, and `arm64` is translated to EC2's `arm64_mac` architecture for Apple silicon.

**Dedicated host logic** landed in #592 (`Support for dedicated hosts`), from #590. `tenancy: host` plus `allocate_dedicated_host` / `deallocate_dedicated_host` allocates and releases hosts as needed. Only hosts tagged `ManagedBy: Test Kitchen` are ever touched, so your own hosts are safe. `.metal` sizes are allocated by exact instance type (a 1:1 host association), everything else by instance family.

@mfortin — to your question, the alternatives you mentioned are supported too, via the `placement` block and `licenses`:

```yaml
driver:
  instance_type: mac1.metal
  placement:
    host_resource_group_arn: arn:aws:resource-groups:us-east-1:123456789012:group/my-mac-hosts
    affinity: host
  licenses:
    - license_configuration_arn: arn:aws:license-manager:us-east-1:123456789012:license-configuration:lic-0123
```

`placement` also accepts `host_id`, `availability_zone`, `group_id` / `group_name`, `partition_number` and `tenancy`.

One caveat worth flagging: until very recently an **unversioned** `macos` platform matched zero images, because the search looked for `amzn2-ec2-macos-*` while real AMIs are named `amzn-ec2-macos-…`. That is fixed in #667. If you tried this and got "Search returned 0 images", that was why — pinning a version (`macos-14`) worked around it.

Reminder for anyone finding this later: dedicated Mac hosts are billed for a **24-hour minimum** from allocation, so `kitchen destroy` does not stop the charge before then.
